app/models/concerns/attachment_jcamp_aasm.rb

```diff
@@ -127,19 +127,25 @@ def generate_att(meta_tmp, addon, to_edit = false, ext = nil)
 
     meta_filename = Chemotion::Jcamp::Gen.filename(filename_parts, addon, ext)
     content_type = ext == 'png' ? 'image/png' : 'application/octet-stream'
-    att = Attachment.new(
-      filename: meta_filename,
-      file_path: meta_tmp.path,
-      attachable_id: attachable_id,
-      attachable_type: 'Container',
-      created_by: created_by,
-      created_for: created_for,
-      content_type: content_type
-    )
+    att = Attachment.children_of(self[:id]).where(filename: meta_filename).take
+    if att.nil?
+      att = Attachment.children_of(self[:id]).new(
+        filename: meta_filename,
+        file_path: meta_tmp.path,
+        created_by: created_by,
+        created_for: created_for,
+        content_type: content_type,
+        key: SecureRandom.uuid
+      )
+    end
+    att.attachment_attacher.attach(File.open(meta_tmp.path, binmode: true))
+    att.attachment_attacher.create_derivatives
     att.save!
     att.set_edited if ext != 'png' && to_edit
     att.set_image if ext == 'png'
     att.set_json if ext == 'json'
+
+    att.update!(attachable_id: attachable_id, attachable_type: 'Container')
```

Comment:
Why do you change the attachable type to Container here? An attachment can also be related to other classes like ResearchPlan. Doing this would destroy existing relations

Reply:
probably something :fish:  but as of now the spectra feature is expected only for  attachments  a dataset.